tap_google_ads/streams.py

```diff
@@ -80,9 +80,43 @@ def build_parameters():
     param_str = ",".join(f"{k}={v}" for k, v in API_PARAMETERS.items())
     return f"PARAMETERS {param_str}"
 
+def generate_where_and_orderby_clause(last_pk_fetched, filter_param, composite_pks):
+    """
+    Generates a where clause based on filter parameter(`key_properties`), and
+    `last_pk_fetched`.
+
+    Example:
+    filter_param = ['id']
+    last_pk_fetched = 1
+    Returns:
+    WHERE id >= 1 ORDER BY id ASC
+
+    """
+    where_clause = ""
+    order_by_clause = ""
+    comparision_operator = ">="
+
+    if not composite_pks:
+        # Exclude equality for the stream which do not have composite primary key.
+        comparision_operator = ">"
```

Comment:
@prijendev Spelling mistake: 'comparison'

Reply:
@prijendev Why are we changing the operator for a single PK case? How do we handle composite PK case in the where clause?

Reply:
Because in single pk case we are sure that no other record will have the same pk. So, we do not want to fetch the last record again. That's why we just keep the`>` operator.

In the case of composite PK, we are storing only a single pk value in the bookmark. So, there might be possible that records with the same single pk value exist with different pk value combinations. That's why for composite_pks we are using a greater than or equal operator.

For example, the `ad_group_criterion` have records as below,

```
{'ad_group_id': 2, 'criterion_id': 1}
{'ad_group_id': 2, 'criterion_id': 4}
{'ad_group_id': 3, 'criterion_id': 4}
{'ad_group_id': 3, 'criterion_id': 2}
```
In this case, the bookmark will be stored as `last_pk_fetched: 3`. (Assuming state will be written after every 3 records). Assume that tap interrupted after 3rd record. Then next time when sync will start, the query with `where` and `order by` clause will be looks like the below,

`WHERE ad_group_id >= 3 ORDER BY ad_group_id ASC`